evm.md

```diff
@@ -461,61 +460,27 @@ The `CallOp` opcodes all interperet their second argument as an address.
     rule <k> #exec [ CO:CallOp     ] => #gas [ CO  , CO  W0 W1 W2 W3 W4 W5 W6 ] ~> CO  W0 W1 W2 W3 W4 W5 W6 ... </k> <wordStack> W0 : W1 : W2 : W3 : W4 : W5 : W6 : WS => WS </wordStack>
 ```
 
-### Helpers
+### Address Conversion
 
--   `#addr` decides if the given argument should be interpreted as an address (given the opcode).
+Some opcodes require accessing elements of the state at different addresses.
+We make sure the given arguments (to be interpreted as addresses) are with 160 bits ahead of time.
 
 ```k
-    syntax InternalOp ::= "#load" "[" OpCode "]"
+    syntax InternalOp ::= "#addr" "[" OpCode "]"
  // --------------------------------------------
-    rule <k> #load [ OP:OpCode ] => #loadAccount #addr(W0) ... </k>
+    rule <k> #addr [ OP:OpCode ] => . ... </k>
          <wordStack> (W0 => #addr(W0)) : WS </wordStack>
-      requires #addr?(OP)
+      requires OP ==K BALANCE
+        orBool OP ==K SELFDESTRUCT
+        orBool OP ==K EXTCODEHASH
+        orBool OP ==K EXTCODESIZE
+        orBool OP ==K EXTCODECOPY
 
-    rule <k> #load [ OP:OpCode ] => #loadAccount #addr(W0) ~> #lookupCode #addr(W0) ... </k>
-         <wordStack> (W0 => #addr(W0)) : WS </wordStack>
-      requires #code?(OP)
-
-    rule <k> #load [ OP:OpCode ] => #loadAccount #addr(W1) ~> #lookupCode #addr(W1) ... </k>
+    rule <k> #addr [ OP:OpCode ] => . ... </k>
          <wordStack> W0 : (W1 => #addr(W1)) : WS </wordStack>
       requires isCallOp(OP) orBool isCallSixOp(OP)
 
-    rule <k> #load [ CREATE ] => #loadAccount #newAddr(ACCT, NONCE) ... </k>
-         <id> ACCT </id>
-         <account>
-           <acctID> ACCT </acctID>
-           <nonce> NONCE </nonce>
-           ...
-         </account>
-
-    rule <k> #load [ OP:OpCode ] => #lookupStorage ACCT W0 ... </k>
-         <id> ACCT </id>
-         <wordStack> W0 : WS </wordStack>
-      requires OP ==K SSTORE orBool OP ==K SLOAD
-
-    rule <k> #load [ OP:OpCode ] => . ... </k>
-      requires notBool (
-        OP ==K CREATE   orBool
-        OP ==K SLOAD    orBool
-        OP ==K SSTORE   orBool
-        isCallOp   (OP) orBool
-        isCallSixOp(OP) orBool
-        #addr?(OP)      orBool
-        #code?(OP)
-      )
-
-    syntax Bool ::= "#addr?" "(" OpCode ")" [function]
- // --------------------------------------------------
-    rule #addr?(BALANCE)      => true
-    rule #addr?(SELFDESTRUCT) => true
-    rule #addr?(EXTCODEHASH)  => true
-    rule #addr?(OP)           => false requires (OP =/=K BALANCE) andBool (OP =/=K SELFDESTRUCT) andBool (OP =/=K EXTCODEHASH)
-
-    syntax Bool ::= "#code?" "(" OpCode ")" [function]
- // --------------------------------------------------
-    rule #code?(EXTCODESIZE)  => true
-    rule #code?(EXTCODECOPY)  => true
-    rule #code?(OP)           => false requires (OP =/=K EXTCODESIZE) andBool (OP =/=K EXTCODECOPY)
+    rule <k> #addr [ OP:OpCode ] => . ... </k> [owise]
```

Comment:
pretty sure the java backend doesn't understand owise on non-function rules.

Reply:
@dwightguth I converted to an explicit side condition.